fs/base.py

```diff
@@ -1219,17 +1220,20 @@ def removetree(self, dir_path):
             dir_path (str): Path to a directory on the filesystem.
 
         """
-        _dir_path = abspath(normpath(dir_path))
         with self._lock:
-            walker = walk.Walker(search="depth")
-            gen_info = walker.info(self, _dir_path)
-            for _path, info in gen_info:
-                if info.is_dir:
-                    self.removedir(_path)
-                else:
-                    self.remove(_path)
-            if _dir_path != "/":
-                self.removedir(dir_path)
+            if self.hassyspath(path):
+                shutil.rmtree(self.getsyspath(path))
+            else:
+                _dir_path = abspath(normpath(dir_path))
+                walker = walk.Walker(search="depth")
+                gen_info = walker.info(self, _dir_path)
+                for _path, info in gen_info:
+                    if info.is_dir:
+                        self.removedir(_path)
+                    else:
+                        self.remove(_path)
+                if _dir_path != "/":
```

Comment:
I think you also need to check that `_dir_path` is not `"/"` with the `shutil.rmtree` branch too, otherwise you may end up deleting the root of the filesystem and potentially preventing new files/directories from being created in that filesystem afterwards.